Prox/Prox/Data/PlacesController.swift

```diff
@@ -139,6 +129,28 @@ class PlacesProvider {
         }
     }
 
+    func filterPlaces(filters: [PlaceFilter]) -> [Place] {
+        let enabledCategories = Set(filters.filter { $0.enabled }.map { $0.categories }.reduce([], +))
+        let toRoots = CategoriesUtil.categoryToRootsMap
+
+        return allPlaces.filter { place in
+            let categories = Set(place.categories.ids.flatMap { toRoots[$0] }.reduce([], +))
+            return !enabledCategories.isDisjoint(with: categories)
+        }
+    }
+
+    /// Applies the current set of filters to all places, setting `displayedPlaces` to the result.
+    /// Callers must acquire a write lock before calling this method!
```

Comment:
> Are we locking whenever we access allPlaces? Add some comments (e.g. at the lock declaration) please!

I think so, but I didn't write this code, so I don't know exactly what `placesLock` is supposed to protect.

I created issue #529, which should clean up this code in general.